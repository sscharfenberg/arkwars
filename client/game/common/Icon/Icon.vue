<script>
    /*******************************************************************************************************************
     * Icon Component
     * these svg icons use a external svg sprite sheet and reference the id in that file
     ******************************************************************************************************************/
    export default {
        props: {
            name: {
                type: String,
                required: true
            }
        },
        computed: {
            iconPath() {
                return `/public/assets/images/icons-game.svg#${this.name}`;
            }
        }
    };
</script>

<template>
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
        <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="" :xlink:href="iconPath"></use>
    </svg>
</template>

<style lang="scss" scoped>
    svg {
        display: inline-block;

        width: 2.4rem;
        height: 2.4rem;

        fill: currentColor;

        &.small {
            width: 2rem;
            height: 2rem;
        }

        &.tiny {
            width: 1.6rem;
            height: 1.6rem;
        }

        &.large {
            width: 3.2rem;
            height: 3.2rem;
        }
    }
</style>

