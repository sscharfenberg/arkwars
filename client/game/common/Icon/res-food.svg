<svg viewBox="0 0 512 512" style="enable-background:new 0 0 512.003 512.003;" xmlns="http://www.w3.org/2000/svg">
    <title>Resource: Food</title>
    <path style="fill:#EDDABF;" d="M241.291,145.259l-53.623-53.623c9.26-9.465,9.202-24.642-0.186-34.03&#10;&#9;c-9.451-9.451-24.774-9.451-34.225,0c-6.206,6.206-8.336,14.944-6.392,22.889c-7.944-1.945-16.683,0.186-22.889,6.392&#10;&#9;c-9.451,9.451-9.451,24.774,0,34.225c9.387,9.387,24.565,9.446,34.03,0.186l53.623,53.623L241.291,145.259z" transform="matrix(1, 0, 0, 1, 3.552713678800501e-15, 7.105427357601002e-15)"/>
    <g transform="matrix(1, 0, 0, 1, 3.552713678800501e-15, 7.105427357601002e-15)">
        <path style="fill:#FFA22B;" d="M360.272,293.901c-30.415,30.415-79.729,30.415-110.145,0&#10;&#9;&#9;c-30.415-30.415-84.123-133.435-53.707-163.851s133.435,23.292,163.851,53.707C390.688,214.172,390.688,263.486,360.272,293.901z"/>
        <path style="fill:#FFA22B;" d="M216.969,160.529l45.598-5.211c117.991-13.485,221.415,78.793,221.415,197.553v36.572&#10;&#9;&#9;c0,16.649-13.497,30.146-30.146,30.146h-271.31c-57.41,0-103.951-46.54-103.951-103.951l0,0&#10;&#9;&#9;C78.577,236.275,138.119,169.54,216.969,160.529z"/>
    </g>
    <path style="fill:#EDDABF;" d="M134.587,179.313l-53.623-53.623c9.26-9.465,9.202-24.642-0.186-34.03&#10;&#9;c-9.451-9.451-24.774-9.451-34.225,0c-6.206,6.206-8.336,14.944-6.392,22.889c-7.944-1.945-16.683,0.186-22.889,6.392&#10;&#9;c-9.451,9.451-9.451,24.774,0,34.225c9.387,9.387,24.565,9.446,34.03,0.186l53.623,53.623L134.587,179.313z" transform="matrix(1, 0, 0, 1, 3.552713678800501e-15, 7.105427357601002e-15)"/>
    <path style="fill:#FFA22B;" d="M253.567,327.954c-30.415,30.415-79.729,30.415-110.145,0&#10;&#9;C113.007,297.539,59.3,194.519,89.716,164.103s133.435,23.292,163.851,53.707S283.983,297.539,253.567,327.954z" transform="matrix(1, 0, 0, 1, 3.552713678800501e-15, 7.105427357601002e-15)"/>
    <path style="fill:#EDDABF;" d="M459.925,461.466H115.999c-23.128,0-41.877-18.749-41.877-41.877l0,0h427.68l0,0&#10;&#9;C501.802,442.717,483.053,461.466,459.925,461.466z" transform="matrix(1, 0, 0, 1, 3.552713678800501e-15, 7.105427357601002e-15)"/>
    <g style="" transform="matrix(1, 0, 0, 1, 3.552713678800501e-15, 7.105427357601002e-15)">
        <path style="fill:#570029;" d="M501.802,409.389h-12.918c3.364-5.887,5.298-12.693,5.298-19.946v-36.572&#10;&#9;&#9;c0-59.508-25.467-116.36-69.87-155.978c-27.083-24.164-59.74-40.582-94.508-48.195c-0.38-0.311-0.778-0.606-1.212-0.866&#10;&#9;&#9;c-24.587-14.733-51.529-26.561-73.916-32.448c-12.584-3.309-23.723-4.819-33.383-4.546l-20.36-20.359&#10;&#9;&#9;c6.637-12.946,4.542-29.302-6.24-40.084c-13.412-13.411-35.236-13.411-48.649,0c-5.409,5.409-8.757,12.244-9.769,19.512&#10;&#9;&#9;c-7.268,1.012-14.103,4.36-19.512,9.769c-6.497,6.498-10.075,15.136-10.075,24.325s3.578,17.828,10.075,24.324&#10;&#9;&#9;c10.782,10.782,27.136,12.878,40.083,6.24l20.394,20.394c-0.042,1.454-0.055,2.932-0.015,4.453&#10;&#9;&#9;c-0.593-0.243-1.186-0.49-1.778-0.727c-18.748-7.546-41.309-14.327-60.836-13.767l-20.384-20.384&#10;&#9;&#9;c6.639-12.946,4.542-29.302-6.24-40.084c-13.412-13.411-35.236-13.412-48.649,0c-5.409,5.409-8.757,12.244-9.769,19.512&#10;&#9;&#9;c-7.268,1.012-14.103,4.36-19.512,9.769c-13.411,13.412-13.411,35.236,0,48.649c10.782,10.782,27.136,12.878,40.084,6.239&#10;&#9;&#9;l20.384,20.385c-0.531,18.56,5.568,39.861,12.654,58.023c-9.815,21.631-14.805,44.7-14.805,68.614&#10;&#9;&#9;c0,38.789,19.452,73.113,49.107,93.751h-43.36c-5.633,0-10.2,4.566-10.2,10.2c0,28.715,23.361,52.077,52.077,52.077h343.927&#10;&#9;&#9;c28.715,0,52.077-23.361,52.077-52.077C512.002,413.956,507.435,409.389,501.802,409.389z M276.025,144.051&#10;&#9;&#9;c-4.868,0.211-9.744,0.577-14.617,1.134l-45.599,5.211c-6.093,0.697-12.127,1.745-18.084,3.099&#10;&#9;&#9;c0.423-6.224,2.002-12.326,5.908-16.232C213.984,126.911,242.181,130.173,276.025,144.051z M165.219,114.084&#10;&#9;&#9;c-3.951-3.951-10.349-3.988-14.345-0.079c-5.458,5.338-14.287,5.289-19.684-0.107c-2.645-2.644-4.101-6.161-4.101-9.9&#10;&#9;&#9;s1.456-7.255,4.101-9.9c3.491-3.491,8.448-4.874,13.251-3.697c3.466,0.849,7.117-0.174,9.638-2.695&#10;&#9;&#9;c2.521-2.521,3.543-6.175,2.696-9.638c-1.177-4.806,0.205-9.761,3.696-13.252c5.46-5.459,14.342-5.459,19.8,0&#10;&#9;&#9;c5.398,5.398,5.446,14.228,0.107,19.685c-3.909,3.995-3.874,10.392,0.079,14.345l17.525,17.525&#10;&#9;&#9;c-3.274,1.747-6.203,3.897-8.771,6.465c-2.562,2.562-4.704,5.502-6.449,8.787L165.219,114.084z M246.355,225.023&#10;&#9;&#9;c26.39,26.39,26.39,69.33,0,95.721c-12.784,12.784-29.781,19.825-47.859,19.825s-35.076-7.041-47.859-19.825&#10;&#9;&#9;c-13.15-13.149-33.659-44.344-47.416-78.521c-13.872-34.466-16.224-60.972-6.291-70.905c4.068-4.068,10.911-6.075,19.948-6.075&#10;&#9;&#9;c13.027,0,30.607,4.175,50.957,12.366C202.012,191.364,233.206,211.872,246.355,225.023z M58.514,148.139&#10;&#9;&#9;c-3.952-3.952-10.349-3.987-14.345-0.079c-5.457,5.337-14.287,5.29-19.684-0.107c-5.459-5.459-5.459-14.341,0-19.799&#10;&#9;&#9;c3.491-3.491,8.445-4.873,13.252-3.697c3.464,0.849,7.117-0.174,9.638-2.695c2.52-2.521,3.542-6.175,2.695-9.638&#10;&#9;&#9;c-1.177-4.806,0.205-9.761,3.696-13.252c5.46-5.46,14.342-5.459,19.8,0c5.398,5.398,5.446,14.227,0.107,19.684&#10;&#9;&#9;c-3.909,3.995-3.874,10.392,0.079,14.345l17.524,17.525c-3.2,1.702-6.143,3.837-8.772,6.465c-2.627,2.627-4.763,5.571-6.465,8.771&#10;&#9;&#9;L58.514,148.139z M88.776,315.638c0-14.476,2.087-28.595,6.2-42.19c12.511,25.012,28.042,48.526,41.235,61.719&#10;&#9;&#9;c17.172,17.171,39.728,25.757,62.284,25.757s45.113-8.586,62.284-25.757l0,0c34.343-34.343,34.343-90.224,0-124.569&#10;&#9;&#9;c-12.051-12.051-32.714-26.053-55.264-37.923c4.167-0.847,8.372-1.528,12.611-2.012l45.599-5.211&#10;&#9;&#9;c53.351-6.098,106.935,10.909,147.006,46.662c40.071,35.751,63.052,87.056,63.052,140.757v36.572&#10;&#9;&#9;c0,10.998-8.948,19.946-19.946,19.946H182.526C130.832,409.389,88.776,367.333,88.776,315.638z M489.919,429.788&#10;&#9;&#9;c-4.255,12.477-16.093,21.478-29.993,21.478H115.999c-13.9,0-25.737-9-29.993-21.478H489.919z"/>
    </g>
</svg>
