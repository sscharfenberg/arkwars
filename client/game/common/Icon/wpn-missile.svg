<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 240 240">
	<g>
		<path fill="#a49c9c" d="M233.538,6.452c-36.354,8.666-62.231,34.854-62.231,34.854l13.67,13.669l0.058,0.058l13.67,13.67 C198.704,68.703,224.871,42.806,233.538,6.452z" />
		<polygon fill="#a49c9c" points="97.92,112.028 33.028,150.084 5.701,177.922 18.356,191.591" />
		<polygon fill="#a49c9c" points="127.981,142.089 89.926,206.981 62.088,234.308 48.419,221.651" />
		<path fill="#a49c9c" d="M161.792,50.264L19.409,192.645l1.259,1.258l-3.49,8.636l20.293,20.294l8.635-3.492l1.258,1.259 L189.747,78.219L161.792,50.264z M33.953,208.4l-1.424-1.422l80.253-80.253l1.423,1.423L33.953,208.4z" />
		<rect fill="#ff0000" x="176.984" y="39.676" transform="matrix(0.7072 -0.707 0.707 0.7072 10.8445 145.0736)" width="7.163" height="39.536" />
	</g>
</svg>
