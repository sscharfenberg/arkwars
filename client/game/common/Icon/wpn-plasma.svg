<svg viewBox="40 -70 512 512" xmlns="http://www.w3.org/2000/svg">
    <title>Plasma</title>
    <desc>"Tesla Turret" by Lorc from game-icons.net</desc>
    <g class="" transform="rotate(-40, 256, 128), scale(1.15)">
        <path d="m 86.25,65.438 c -16.888,0.292 -31.98,8.265 -43.625,20.53 -13.8,14.54 -22.688,35 -22.688,55.97 0,23.592 10.85,46.488 27.75,60.968 -16.63,20.026 -23.824,40.488 -23.406,58.938 0.462,20.306 10.22,37.75 24.845,48.5 14.624,10.75 34.166,14.9 53.656,9.594 18.535,-5.046 36.713,-18.573 51.032,-41.563 h 102.313 v -64.47 l -156.313,22.19 v 42.28 H 131 c -10.582,13.32 -22.16,20.546 -33.125,23.53 -14.113,3.843 -27.553,0.825 -37.688,-6.624 -10.134,-7.448 -16.917,-19.23 -17.25,-33.874 -0.312,-13.763 5.157,-30.422 20.657,-48.312 9.81,4.438 20.733,6.306 32.218,4.687 l 4,-0.56 185.625,-26.345 v -39.47 h 37.532 v 34.126 l 18.686,-2.655 v -31.47 h 34.28 l 0.005,26.592 18.687,-2.625 v -23.97 h 27.938 c 4,15.145 17.786,26.314 34.187,26.314 19.532,0 35.375,-15.844 35.375,-35.376 0,-19.532 -15.843,-35.344 -35.375,-35.344 -16.19,0 -29.804,10.872 -34,25.72 h -28.125 v -20.47 l -18.688,-2.97 v 23.44 h -34.28 v -28.907 l -18.688,-2.97 v 31.876 H 285.438 V 94.843 L 273.22,92.905 95.81,66.062 c -3.24,-0.494 -6.435,-0.68 -9.562,-0.624 z m 119.938,44.718 -5,21.813 62.218,11.843 -82.437,27.562 5,-21.813 -62.22,-11.812 82.438,-27.594 z" />
    </g>
</svg>
