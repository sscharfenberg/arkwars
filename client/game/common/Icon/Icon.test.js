import {mount} from "vue-test-utils";
import Icon from "./Icon.vue";

describe("Icon.test.js", () => {
    let cmp;

    beforeEach(() => {
        cmp = mount(Icon, {propsData: {name: "sync", size: 3}});
    });

    it("has the expected prop", () => {
        expect(cmp.vm.name).toEqual("sync");
    });

    it("has the expected html structure", () => {
        expect(cmp.vm.$el).toMatchSnapshot();
    });

    it('uses class="large" size=3', () => {
        expect(cmp.find("svg").attributes().class).toMatch(/large/);
    });
});
