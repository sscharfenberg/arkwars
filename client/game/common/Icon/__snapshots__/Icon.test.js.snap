// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.test.js has the expected html structure 1`] = `
<svg
  aria-hidden="true"
  class="large"
  xmlns="http://www.w3.org/2000/svg"
>
  <use
    xlink:href="#g-sync"
  />
</svg>
`;
